common/test/acceptance/tests/lms/test_teams.py

```diff
@@ -1439,6 +1439,22 @@ def test_join_team(self):
         self.teams_page.click_all_topics()
         self.verify_my_team_count(1)
 
+    def test_join_team_from_teams_page(self):
```

Comment:
That does seem like a better approach. I've updated the test to navigate to the team page, rather than visiting it directly, and verified that it does fail without the bugfix.